plugins/core/gov.nasa.jpl.imce.oml.viewpoint/src/gov/nasa/jpl/imce/oml/viewpoint/ConceptUsageDiagramService.xtend

```diff
@@ -0,0 +1,438 @@
+package gov.nasa.jpl.imce.oml.viewpoint
+
+import gov.nasa.jpl.imce.oml.model.terminologies.Aspect
+import gov.nasa.jpl.imce.oml.model.terminologies.Concept
+import gov.nasa.jpl.imce.oml.model.terminologies.Entity
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityExistentialRestrictionAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityRelationship
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityRestrictionAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.EntityUniversalRestrictionAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.ReifiedRelationship
+import gov.nasa.jpl.imce.oml.model.terminologies.SpecializationAxiom
+import gov.nasa.jpl.imce.oml.model.terminologies.TerminologyBoxStatement
+import java.util.AbstractMap.SimpleEntry
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.HashSet
+import java.util.LinkedList
+import java.util.List
+import java.util.Map
+import java.util.Map.Entry
+import java.util.Queue
+import java.util.Set
+import org.eclipse.sirius.diagram.DDiagram
+import org.eclipse.sirius.diagram.DSemanticDiagram
+import gov.nasa.jpl.imce.oml.model.terminologies.AspectSpecializationAxiom
+
+class ConceptUsageDiagramService {

```

Comment:
Explain what this diagram is intended to show in terms of queries on the model:

what are the boxes?
what are the lines?